presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -508,12 +510,14 @@ protected Node visitShowCreate(ShowCreate node, Void context)
                 ConnectorTableMetadata connectorTableMetadata = metadata.getTableMetadata(session, tableHandle.get()).getMetadata();
 
                 Map<String, PropertyMetadata<?>> allColumnProperties = metadata.getColumnPropertyManager().getAllProperties().get(tableHandle.get().getConnectorId());
-
+                Set<String> reservedIdentifiers = ReservedIdentifiers.reservedIdentifiers();
```

Comment:
Looks like `ReservedIdentifiers.reservedIdentifiers();
Write Preview` builds a new set on every call. This feel inefficient.

Reply:
Thanks for bringing it up. I actually thought about this and since the number of entries its processing is ~200, I thought this is ok for show create flow and we avoid adding any additional memory footprint (if we plan to cache it). 

But ReservedIdentifiers::reservedIdentifiers() returns total 69 entries which is not a lot and can be cached in ReservedIdentifiers and return the cached result in ReservedIdentifiers->reservedIdentifiers() instead of reprocessing. WDYT?

Reply:
I think caching would be helpful. It will make this command faster and use less memory and therefore less GC.